src/basic/stat-util.c

```diff
@@ -224,3 +224,13 @@ int fd_is_temporary_fs(int fd) {
 
         return is_temporary_fs(&s);
 }
+
+int path_is_temporary_fs(const char *path) {
+        _cleanup_close_ int fd = -1;
+
+        fd = open(path, O_RDONLY|O_CLOEXEC|O_NOCTTY);
```

Comment:
so, this works fine even unprivileged.I'll add a follow-commit to the set, that ports this over and does hte same for the existing path_check_fstype(), and adds some tests for it.